From fac0868b13817f19f11de0b30e4f738be38299e1 Mon Sep 17 00:00:00 2001
From: Joey <joey@lineageos.org>
Date: Wed, 28 Mar 2018 21:11:16 +0200
Subject: [PATCH 06/11] Dialer: comply with EU's GDPR

Disable lookup by default and add a disclaimer for the feature

Change-Id: If7a181952304dbaee736762bdfd5819eddc5f89b
Signed-off-by: Joey <joey@lineageos.org>
---
 java/com/android/dialer/lookup/LookupSettings.java         | 6 +++---
 java/com/android/dialer/lookup/LookupSettingsFragment.java | 6 +++---
 java/com/android/dialer/lookup/res/values/cm_strings.xml   | 3 +++
 java/com/android/dialer/lookup/res/xml/lookup_settings.xml | 1 +
 4 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/java/com/android/dialer/lookup/LookupSettings.java b/java/com/android/dialer/lookup/LookupSettings.java
index bed395dee..770d6c7e8 100644
--- a/java/com/android/dialer/lookup/LookupSettings.java
+++ b/java/com/android/dialer/lookup/LookupSettings.java
@@ -53,17 +53,17 @@ public final class LookupSettings {
 
     public static boolean isForwardLookupEnabled(Context context) {
         return LineageSettings.System.getInt(context.getContentResolver(),
-                LineageSettings.System.ENABLE_FORWARD_LOOKUP, 1) != 0;
+                LineageSettings.System.ENABLE_FORWARD_LOOKUP, 0) != 0;
     }
 
     public static boolean isPeopleLookupEnabled(Context context) {
         return LineageSettings.System.getInt(context.getContentResolver(),
-                LineageSettings.System.ENABLE_PEOPLE_LOOKUP, 1) != 0;
+                LineageSettings.System.ENABLE_PEOPLE_LOOKUP, 0) != 0;
     }
 
     public static boolean isReverseLookupEnabled(Context context) {
         return LineageSettings.System.getInt(context.getContentResolver(),
-                LineageSettings.System.ENABLE_REVERSE_LOOKUP, 1) != 0;
+                LineageSettings.System.ENABLE_REVERSE_LOOKUP, 0) != 0;
     }
 
     public static String getForwardLookupProvider(Context context) {
diff --git a/java/com/android/dialer/lookup/LookupSettingsFragment.java b/java/com/android/dialer/lookup/LookupSettingsFragment.java
index 8ec1028ec..bf0272967 100644
--- a/java/com/android/dialer/lookup/LookupSettingsFragment.java
+++ b/java/com/android/dialer/lookup/LookupSettingsFragment.java
@@ -107,11 +107,11 @@ public class LookupSettingsFragment extends PreferenceFragment
     private void restoreLookupProviderSwitches() {
         final ContentResolver cr = getActivity().getContentResolver();
         mEnableForwardLookup.setChecked(LineageSettings.System.getInt(cr,
-                LineageSettings.System.ENABLE_FORWARD_LOOKUP, 1) != 0);
+                LineageSettings.System.ENABLE_FORWARD_LOOKUP, 0) != 0);
         mEnablePeopleLookup.setChecked(LineageSettings.System.getInt(cr,
-                LineageSettings.System.ENABLE_PEOPLE_LOOKUP, 1) != 0);
+                LineageSettings.System.ENABLE_PEOPLE_LOOKUP, 0) != 0);
         mEnableReverseLookup.setChecked(LineageSettings.System.getInt(cr,
-                LineageSettings.System.ENABLE_REVERSE_LOOKUP, 1) != 0);
+                LineageSettings.System.ENABLE_REVERSE_LOOKUP, 0) != 0);
     }
 
     private void restoreLookupProviders() {
diff --git a/java/com/android/dialer/lookup/res/values/cm_strings.xml b/java/com/android/dialer/lookup/res/values/cm_strings.xml
index 28af84428..fed7c0067 100644
--- a/java/com/android/dialer/lookup/res/values/cm_strings.xml
+++ b/java/com/android/dialer/lookup/res/values/cm_strings.xml
@@ -30,4 +30,7 @@
     <string name="forward_lookup_provider_title">Forward lookup provider</string>
     <string name="people_lookup_provider_title">People lookup provider</string>
     <string name="reverse_lookup_provider_title">Reverse lookup provider</string>
+
+    <!-- Disclaimer -->
+    <string name="lookup_disclaimer">Lookups may send queries over a secure protocol (https) to remote websites to gather information. The query may include the other party\'s phone number or the search query</string>
 </resources>
diff --git a/java/com/android/dialer/lookup/res/xml/lookup_settings.xml b/java/com/android/dialer/lookup/res/xml/lookup_settings.xml
index 5149b6324..006345f52 100644
--- a/java/com/android/dialer/lookup/res/xml/lookup_settings.xml
+++ b/java/com/android/dialer/lookup/res/xml/lookup_settings.xml
@@ -65,5 +65,6 @@
         android:summary="%s"
         android:persistent="false" />
 
+    <Preference android:summary="@string/lookup_disclaimer" />
 </PreferenceScreen>
 
-- 
2.17.1

